subsys/nrf_security/src/drivers/nrf_oberon/CMakeLists.txt

```diff
@@ -66,4 +66,43 @@ if (COMPILE_PSA_APIS)
 	-Wno-uninitialized
 	-Wno-maybe-uninitialized
   )
+
+elseif(CONFIG_NRF_SECURITY_LEGACY_AND_PSA)
+  # When legacy and PSA are enabled, we need to include all the oberon drivers
+  # since we don't have PSA_NEED_* symbols in Kconfig.
+  list(APPEND src_crypto_oberon
+       ${drivers_path}/oberon_helpers.c
+       ${drivers_path}/oberon_ecdh.c
+       ${drivers_path}/oberon_ecdsa.c
+       ${drivers_path}/oberon_ec_keys.c
+       ${drivers_path}/oberon_jpake.c
+       ${drivers_path}/oberon_spake2p.c
+       ${drivers_path}/oberon_srp.c
+       ${drivers_path}/oberon_rsa.c
+       ${drivers_path}/oberon_key_management.c
+       ${drivers_path}/oberon_aead.c
+       ${drivers_path}/oberon_key_derivation.c
+       ${drivers_path}/oberon_mac.c
+       ${drivers_path}/oberon_cipher.c
+       ${drivers_path}/oberon_hash.c
+       ${drivers_path}/oberon_key_agreement.c
+       ${drivers_path}/oberon_pake.c
+       ${drivers_path}/oberon_asymmetric_signature.c
+       ${drivers_path}/oberon_asymmetric_encrypt.c
+       ${drivers_path}/oberon_ctr_drbg.c
+       ${drivers_path}/oberon_hmac_drbg.c)
+
+
+  target_sources(${mbedcrypto_target} PRIVATE ${src_crypto_oberon})
+
+  # Turn off warnings that Oberon are systematically
+  # triggering. Oberon is testing this deliverable before we get it so
+  # we don't need to re-test it with gcc warnings.
+  target_compile_options(
+	${mbedcrypto_target}
+	PRIVATE
+	-Wno-uninitialized
+	-Wno-maybe-uninitialized
+  )
```

Comment:
This could be outside of the if/elseif block as we do the same no what branch taken,